### PR TITLE
CONTRIBUTING: add style guide

### LINKED ISSUE #781: readme: document canonical sample style
@kurtisvg, is this something you could help with? It would help contributors know how to write samples and would be nice to be able to link to during code review (each part with a separate anchor that's easy to link to, like Go code review comments).

### PR DESCRIPTION
**Pull request by tbpg**
Fixes #781.

### CODE REVIEW

**Comment by tbpg** [comment]
Failure caused by one of the four projects hitting a service unavailable error.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,19 +12,240 @@
...
+all Go code in this repo.
+
+The following style guidelines are specific to writing Go samples.
+
+## One file per sample
+
+Each sample should be in its own file so the imports used by the sample can
+be included in the region tag.
+
+## File paths
```
[the remark this answers is not in the document]
Comment (tbpg) [reply CONTRIBUTING.md]: Done. Did "Sample file name and directory."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,19 +12,240 @@
...
+client, err := foo.NewClient(...)
+- if err != nil {
+- 	log.Fatal(err)
+- }
++ if err != nil {
++ 	return fmt.Errorf("foo.NewClient: %v", err)
++ }
+```
+
+## Error messages
```
[the remark this answers is not in the document]
Comment (tbpg) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,19 +12,282 @@
...
+
+There should be as few function arguments as possible. An `io.Writer` and
+project ID are the most common. If you need additional arguments (for example,
+the ID of a resource to get or delete), there should be an example value in the
+body of the sample function.
+
+```go
+// delete deletes the resource identified by resourceID.
+func delete(w io.Writer, resourceID string) error {
+	// resource := fmt.Sprintf("/projects/my-project/resources/%s", resourceID)
```
Comment (broady) [inline CONTRIBUTING.md]: i think we call these "name" - do we need to add common identifiers to the style guide, too?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,19 +12,282 @@
...
+	client, err := foo.NewClient()
```
Comment (broady) [inline CONTRIBUTING.md]: NewClient probably takes a ctx here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -12,19 +12,282 @@
...
++		return fmt.Errorf("Delete: %v", err)
++	}
+	return nil
+}
+```
+
+
+## Imports
+
+Imports should be added and sorted by `goimports`. There should be two groups,
```
Comment (broady) [inline CONTRIBUTING.md]: at least two groups. i think goimports adds more sometimes

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -12,19 +12,282 @@
...
+https://godoc.org/github.com/GoogleCloudPlatform/golang-samples. If there are
+many files/samples in the package, it's common to create a `doc.go` file that
+only has the package comment. The comment should start with
+`Package packagename`.
+
+Functions should have comments starting with the name of the function (with the
+same capitalization, even if it's lower case).
+
+```go
+// package foo contains samples for Foo.
```
Comment (broady) [inline CONTRIBUTING.md]: "Package foo" - sentence case

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -12,19 +12,282 @@
...
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"golang.org/x/exp/rand"
+)
+```
+
+## Comment functions and packages
```
Comment (broady) [inline CONTRIBUTING.md]: any guideline on naming for packages and their directory names?

Comment (tbpg) [reply CONTRIBUTING.md]: Added ("Sample package name, file name, and directory").

**Pull request merged**